### ISSUE TITLE
setResource is broken when the resource contains only 1 data element

### ISSUE THREAD

**Issue by arthens** [blongden/hal#51]
```php
$resource = new Nocarrier\Hal('/post/1',[/*some data here*/]);
$author = new Nocarrier\Hal('/user/1', [
    'name' => 'John',
]);
$resource->setResource('user', $author);

$resource->asJson();
```

Will fail with `Notice: Undefined offset: 0` [here](https://github.com/blongden/hal/blob/master/src/Nocarrier/HalJsonRenderer.php#L158):

```php
if (count($embedded) === 1 && !in_array($rel, $resource->getArrayResourceRels())) {
    $embedded = $embedded[0];
}
```

This fails because:
- $author only contains 1 field, which triggers the condition `count($embedded) === 1`
- `setResource` does not register refs in `arrayResourceRels`, which triggers the condition `!in_array($rel, $resource->getArrayResourceRels())`

I don't fully understand the logic here, so I'm not sure what the solution is, but it was extremely confusing to find out that having a resource with a single field was not supported. I had to add a second (unnecessary) field to make it work.

**Comment by blongden** [comment]
The reason for this is to (by default) prevent a single resource from being added as an array. I'll try to reproduce the issue and provide a fix for it.

**Comment by blongden** [comment]
Yeah ok - having actually _read_ the code this is the same bug that was fixed in #49. But this was not merged into master. I have created a new release, 0.9.12 that contains the fix. If you update your composer.json you should be good.

**Comment by arthens** [comment]
Thanks!
